File: spring-data-requery/src/main/java/org/springframework/data/requery/converters/DateTimeToTimestampConverter.java
package org.springframework.data.requery.converters;

import io.requery.Converter;
import org.joda.time.DateTime;

import javax.annotation.Nullable;
import java.sql.Timestamp;

/**
 * Convert joda-time {@link DateTime} to {@link Timestamp}
 *
 * @author debop
 */
public class DateTimeToTimestampConverter implements Converter<DateTime, Timestamp> {

    @Override
    public Class<DateTime> getMappedType() {
        return DateTime.class;
    }

    @Override
    public Class<Timestamp> getPersistedType() {
        return Timestamp.class;
    }

    @Nullable
    @Override
    public Integer getPersistedSize() {
        return null;
    }

    @Override
    public Timestamp convertToPersisted(DateTime value) {
        return (value != null) ? new Timestamp(value.getMillis()) : null;
    }

    @Override
    public DateTime convertToMapped(Class<? extends DateTime> type, @Nullable Timestamp value) {
        return (value != null) ? new DateTime(value.getTime()) : null;
    }
}

File: spring-data-requery-kotlin/src/main/kotlin/org/springframework/data/requery/kotlin/converters/DateTimeToTimestampConverter.kt
package org.springframework.data.requery.kotlin.converters

import io.requery.Converter
import org.joda.time.DateTime
import java.sql.Timestamp

/**
 * Convert [DateTime] to [Timestamp]
 *
 * @author debop
 * @since 18. 7. 2
 */
class DateTimeToTimestampConverter: Converter<DateTime, Timestamp> {

    override fun getPersistedType(): Class<Timestamp> = Timestamp::class.java
    override fun getMappedType(): Class<DateTime> = DateTime::class.java
    override fun getPersistedSize(): Int? = null

    override fun convertToMapped(type: Class<out DateTime>?, value: Timestamp?): DateTime? {
        return value?.let { DateTime(it.time) }
    }

    override fun convertToPersisted(value: DateTime?): Timestamp? {
        return value?.let { Timestamp(it.millis) }
    }


}